tools/istio-iptables/pkg/dependencies/implementation.go

```diff
@@ -29,18 +56,54 @@ func (r *RealDependencies) execute(cmd string, redirectStdout bool, args ...stri
 	fmt.Printf("%s %s\n", cmd, strings.Join(args, " "))
 	externalCommand := exec.Command(cmd, args...)
 	externalCommand.Stdout = os.Stdout
+
+	var stderr bytes.Buffer
 	// TODO Check naming and redirection logic
 	if !redirectStdout {
-		externalCommand.Stderr = os.Stderr
+		externalCommand.Stderr = &stderr
+	}
+
+	err := externalCommand.Run()
+	// TODO Check naming and redirection logic
+	if err != nil && !redirectStdout {
+		stderrStr := stderr.String()
+
+		if isXTablesCmd(cmd) {
+			// Transform to xtables-specific error messages with more useful and actionable hints.
+			stderrStr = transformToXTablesErrorMessage(stderrStr, err)
+		}
+
+		// Print stderr to os.Stderr by default.
+		fmt.Fprintln(os.Stderr, stderrStr)
+	}
+
+	return err
+}
+
+// isXTablesCmd checks whether the input is a xtables command.
+func isXTablesCmd(cmd string) bool {
+	return cmd == constants.IPTABLES || cmd == constants.IP6TABLES
```

Comment:
So this won't work for `iptables-restore` / `ip6tables-restore`? Those are the commands that are actually run.

Reply:
Right, double confirmed with iptables source code [1][2] and this should work for all supported xtables cmds including `xtables-restore` and `xtables-save` as well.
Thanks a lot!

[1] https://git.netfilter.org/iptables/tree/iptables/iptables-restore.c?h=v1.8.4
[2] https://git.netfilter.org/iptables/tree/iptables/iptables-save.c?h=v1.8.4